docs/Deployment/RaspberryPi4.md

```diff
@@ -0,0 +1,328 @@
+chmod +x btcpayserver-install.sh
+. btcpayserver-install.sh
+```
+
+After the initial setup is complete open a browser on another computer and go to `btcpay.local`.
+
+:::tip
+Your installation is done and your node should have started synching.
+For a Bitcoin full node the initial block download takes approximately 40 hours after install.
+:::
+
+If you are curious, here are the details of what the installation scripts above do …
+
+## Detailed Step by Step Instructions
+
+These are the steps that follow after the general setup process outlined in the [quickstart instructions](#quickstart).
+
+:::tip NOTE
+The following steps require you to be the root user.
+
+```bash
+sudo su -
+```
+:::
+
+### Upgrade the OS packages to latest
+
+```bash
+apt update && apt upgrade -y && apt autoremove
+```
+
+### Configuring the storage
+
+We recommend to disable swap to prevent burning out your SD card:
+
+```bash
+dphys-swapfile swapoff
+dphys-swapfile uninstall
+update-rc.d dphys-swapfile remove
+systemctl disable dphys-swapfile
+```
+
+Partition your SSD:
+
+```bash
+fdisk /dev/sda
+# type 'p' to list existing partitions
+# type 'd' to delete currently selected partitions
+# type 'n' to create a new partition
+# type 'w' to write the new partition table and exit fdisk
+```
+
+Format the new partition on your SSD:
+
+```bash
+mkfs.ext4 /dev/sda1
+```
+
+Configure the SSD partition to auto-mount at bootup:
+
+```bash
+mkfs.ext4 /dev/sda1
+mkdir /mnt/usb
+UUID="$(sudo blkid -s UUID -o value /dev/sda1)"
+echo "UUID=$UUID /mnt/usb ext4 defaults,noatime,nofail 0 0" | sudo tee -a /etc/fstab
+mount -a
+```
+
+While you’re editing `/etc/fstab` add a RAM filesystem for logs (optional).
+This is also to prevent burning out your SD card too quickly:
+
+```bash
+echo 'none /var/log tmpfs size=10M,noatime 0 0' >> /etc/fstab
+```
+
+### Install Docker
+
+```bash
+apt install apt-transport-https ca-certificates curl gnupg lsb-release -y
+curl -fsSL https://download.docker.com/linux/debian/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/debian \
+  $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
+apt update
+apt -y install docker-ce docker-ce-cli containerd.io
+```
+
+### Create mount for Docker volumes
+
+```bash
+rm -rf /var/lib/docker
+mkdir -p /var/lib/docker
+mount --bind /mnt/usb /var/lib/docker
+echo "/mnt/usb /var/lib/docker none bind,nobootwait 0 2" >> /etc/fstab
+systemctl restart docker
+```
+
+### Configuring the firewall
+
+Install a firewall and allow SSH, HTTP, HTTPS, Bitcoin, and Lightning:
+
+```bash
+apt install -y ufw
+ufw default deny incoming
+ufw default allow outgoing
+```
+
+This command allows SSH connections from internal networks only:
+
+```bash
+ufw allow from 10.0.0.0/8 to any port 22 proto tcp
```

Comment:
You may consider to open port 53 too, for those who will enable pihole. 
I use eth0 and wlan0 to filter on the local network, but I had to rename the docker bridged interface as br-btcpayserver otherwise the ssh connections from btcpayserver container wouldn't work.
See my ufw configuration at https://github.com/gradientskier/btcpayserver-docker/blob/arm64-bullseye-prod/Ansible/roles/ufw/templates/expected_ufw_output.txt.tmpl

Reply:
As the Docker fragments are optional and additional, I would say we should focus on the minimum here and rather document this as an additional step in the [Pi-Hole docs](https://github.com/btcpayserver/btcpayserver-docker/edit/master/docs/pihole.md). Can you add it there?